tests/test_client.py

```diff
@@ -18,12 +18,12 @@ class TestClient(unittest.TestCase):
 
     def test_repo_init(self):
         with tempfile.TemporaryDirectory() as tmpdir:
-            Client("user0", tmpdir, "fake_hash", init_repo=True)
+            Client("user0", tmpdir, "fake_hash", tmpdir+"/fake_bin", init_repo=True)
```

Comment:
Why have you updated all these tests Client args? I think you forgot to change these back for `test_client` and `test_state`.